frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.tsx

```diff
@@ -125,21 +133,22 @@ function ActionCreatorComponent({
     if (isNew) {
       setShowSaveModal(true);
     } else {
-      update({
-        id: question.id(),
-        name: question.displayName() ?? "",
-        description: question.description() ?? null,
-        model_id: defaultModelId as number,
-        formSettings: formSettings as ActionFormSettings,
-        question,
-      });
+      const action = convertQuestionToAction(question, formSettings);
+      handleUpdateAction({ ...action, model_id: defaultModelId as number });
       onClose?.();
     }
   };
 
-  const handleSave = (action: WritebackQueryAction) => {
-    const actionCard = convertActionToQuestionCard(action);
-    setQuestion(question.setCard(actionCard));
+  const handleCreate = async (values: any) => {
```

Comment:
Can we set a type here? If I'm not mistaken it would be `FormValues` from `CreateActionForm.tsx`

Reply:
Oh, totally! Temporarily used `any` and here I am forgetting to fix it before pushing :(